js/main.js

```diff
@@ -98,3 +149,18 @@ document.addEventListener('DOMContentLoaded', function (event) {
     swapView('profile');
   }
 });
+var $modal = document.querySelector('#modal');
+
+document.addEventListener('click', function (event) {
+  if (event.target.tagName === 'A' && event.target.className !== 'header-profile') {
+    swapView('edit-profile');
+  } else if (event.target.className === 'header-profile' && data.profile.username.length !== 0) {
+    swapView('profile');
+  } else if (event.target.className === 'header-profile' && data.profile.username.length === 0) {
+    document.querySelector('.container').classList.add('modal-check');
+    document.querySelector('#modal').classList.remove('hidden');
+  } else if (event.target.className === 'x') {
```

Comment:
I would make this its own separate event handler, since it doesn't match the focus of the rest of the function. This function is meant to be for checking if a user needs to move to a different view, but this portion is checking if the user is closing the modal. This would be best being an event handler attached directly to the `x` element 👍